bin/build/build.sh

```diff
@@ -3,6 +3,31 @@
 declare htaccess_config_default="htaccess.conf";
 declare htaccess_output_default="./.htaccess"
 declare repo_root="$(cd "$(dirname "$0")" && cd ../../ && pwd)"
+declare temp_directory
+declare htaccess_output_tmp
+
+# ----------------------------------------------------------------------
+# | File system                                                        |
+# ----------------------------------------------------------------------
+
+prepareTempDirectory() {
+    temp_directory="$(mktemp -d ${TMPDIR:-/tmp/}server-configs-apache.XXXXXXXXXXXX)"
+    htaccess_output_tmp="${temp_directory}/htaccess.tmp"
+
+    if [ ! -d "${temp_directory}" ]; then
+        print_error "Creating temp directory failed."
+        exit 1
+    fi
+}
+
+# Default Exit or SIGINT(2) handler
+trapCleanupTempDir() {
+    [ -d "${temp_directory}" ] \
+    && rm -Rf "${temp_directory}" \
```

Comment:
I don't like at all your last commit...

This commit adds too much extra complexity, and this command is really much more risky than having a failed build.

Let user handle their filesystem and temp build files. 😅

Reply:
At first, thank you for your review. It must be hard work to look through dozens od commits... 

Please allow me to plead: Project roots, especially on live environments, are not the place where to risk cluttered file system. See, it is widely-used practice to not checkout .htaccess with the repo, and to maintain the .htaccess on each enviroment individually (for example: rewrite_base directives). So developers will likely checkout their conf file instead, and build their production .htaccess with our build tool. If only one small thing fails, the web app can be considered broken for at least minutes. This is why I'd prefer a “sandbox” for building. 

You indeed are right, the trap EXIT SIGINT stuff is complex (and surely not seen very often). But e.g. on shared hosting environments I'd prefer to not have temp builds lying around. This was my reason for “use temp, but tidy up after all”. 

If we omitted this functionality, well, we would have to: 

- Check for existing .htaccess file first
- Create backup
- Try build process
- If it fails, restore .htaccess from backup.

This steps alone relocate the “too-much complexity” back into main(), so no use here.

And: currently, the create function 'exits 1' without discussion when something fails. In contradiction, backup restauration implies returning back to main function, adding even more complexity in return code evaluation.

Of course, users should have their filesystem on radar. But all this boilerplate stuff can be tempting to not have. Some people might even confuse it with “not need to” ;-)

Reply:
I still see your point, @LeoColomb. What about a reasonable compromise? Commit #56d1eeb decreases output verbosity and removes redundant directory checks.

Reply:
Finally, I found a way to go without the trap function, and to just work with a tmp file right beneath the final .htaccess. In theory this has the flaw that in case of error exits, after mkdir -p, there might be a newly created path/to/custom/ directory which is in fact empty. 

Anyhow, I ran into some trouble with Travis. Please see new comment below.

Reply:
> This is why I'd prefer a “sandbox” for building.

Sorry for my late reply, but I still think what you call sandbox for build must be managed by the user, not by the building tool.

* More and more deployment pipelines required build step to be separated from the deploy step, even more when using CI/CD.
* This add some system-relative complication, I mean are you sure that the directories used outside of the repo's one are always present?
* This may generate more issues regarding deployment, with users expecting a perfect deployment script, which is not our goal and where I definitely prefer adding a warning notice.

IMHO, a check step before clean/build is more relevant. Checking if all files are available and readable.

Reply:
Thank you for rour reply. It hasn't been too long since, and btw. my three-day-holidays are over, so my commit frequency will melt down to a average spare time level again ;-)

Regarding our discussion, these are arguments I can follow. (External temp dir usage has already been removed with one *cough* of the latest commits.)

- I like the idea with a before-check if the files (or the target directory) in question can be written at all. “…adding a warning notice” – we could check before `mkdir -p`, if $PWD is writeable.
- Can we safely assume the dir created with `make -p` is writeable? I think so.
- Raising an error when .htaccess exists is not necessary IMHO; Here I would prefer the backup created, together with a notice on this (like it is done now), because creating/updating an htaccess is exactly what this script is supposed to do. Users will know that there may exist one. Perhaps just a matter of taste?